Automattic/simplenote-ios
# History slider

### Info
This PR replaces old note history component with a new history slider as described in #762.
Along with the main task the following improvements were made:
* Support rotation while history slider is shown
* Allow scrolling the content of the note editor
* History slider component waits till all versions are loaded before showing the slider
* Error message is shown inline inside the card instead of in a popup alert view

### Pending changes
The following changes are expected in the next PR:

* Put node editor in read only mode and disable interactions with navigation bar while history slider is visible
* Add voice over support 
* Update dismiss button according to mockups
* Support theme change notifications and improve light / dark mode support
* Add unit tests

### Test
Common steps:
1. Open a note
2. Open note options
3. Tap on the history button

What can be tested after that:
* If no internet connection, inline error message is shown
* If current version is selected, restore button is disabled
* On devices with haptic feedback, dragging a slider triggers haptic feedback
* Screen can be rotated to landscape / portrait 
* Selecting a version updates note editor with that version
* Selecting a version and dismissing a history slider restores editor to the current version
* Selecting a version and saving it should persist the changes


<p float="left">
<img src="https://user-images.githubusercontent.com/54751/86618362-12979380-bfb9-11ea-8778-780d64ed8d74.png" width="320" />
<img src="https://user-images.githubusercontent.com/54751/86618366-14f9ed80-bfb9-11ea-96b6-8790cea85626.png" width="320" />
</p>
